curvlinops/kfac.py

```diff
@@ -546,10 +560,23 @@ def _compute_loss_and_backward(self, output: Tensor, y: Tensor):
             loss = self._loss_func(output, y)
             loss.backward()
 
+        elif self._fisher_type == "forward-only":
+            # Since FOOF sets the gradient covariance Kronecker factors to the identity,
+            # we don't need to do a backward pass. See https://arxiv.org/abs/2201.12250.
+            # We choose to set the gradient covariance to the identity explicitly for
+            # the sake of simplicity, such that the rest of the code here and for
+            # `KFACInverseLinearOperator` does not have to be adapted. This could be
+            # changed to decrease the memory costs.
+            for mod_name, param_pos in self._mapping.items():
+                param = self._params[next(iter(param_pos.values()))]
+                self._gradient_covariances[mod_name] = eye(
+                    param.shape[0], dtype=param.dtype, device=self._device
+                )
```

Comment:
Question: Can we simplify L571 or add a comment what it does? It seems to determine the output dimension, but I got slightly confused while reading it.